### PR TITLE
PILOT-1993: Convert utils into email client dependency

### PR DESCRIPTION
**Pull request by vmoshynskyi**
## Summary

Convert utils into email client dependency

## JIRA Issues

PILOT-1993

## Type of Change

- [x] Refactor or reformatting

## Testing

Are there any new or updated tests to validate the changes?

- [x] Yes
- [ ] No

### CODE REVIEW

File: notification/components/email/email_client.py [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+_logger = LoggerFactory(
+    __name__,
+    level_default=settings.LOG_LEVEL_DEFAULT,
+    level_file=settings.LOG_LEVEL_FILE,
+    level_stdout=settings.LOG_LEVEL_STDOUT,
+    level_stderr=settings.LOG_LEVEL_STDERR,
+).get_logger()
+
+
+class EmailClient:
```
Comment (afmam) [inline notification/components/email/email_client.py:43]: Missing docstrings and type hints for all methods in this class

File: notification/components/email/schemas.py [hunk 2]
```diff
@@ -50,7 +51,7 @@ def set_code(self, code):
         self.code = code
 
 
-class POSTEmail(BaseModel):
+class POSTEmail(BaseSchema):
```
Comment (afmam) [inline notification/components/email/schemas.py:54]: Missing docstring

File: notification/components/email/schemas.py [hunk 3]
```diff
@@ -30,7 +31,7 @@ class EAPIResponseCode(Enum):
     to_large = 413
 
 
-class APIResponse(BaseModel):
+class APIResponse(BaseSchema):
```
Comment (afmam) [inline notification/components/email/schemas.py:34]: Missing docstring

File: notification/components/email/views.py [hunk 4]
```diff
@@ -42,58 +38,24 @@
...
-    client.quit()
-
-
 @router.post('/', response_model=POSTEmailResponse, summary='Send emails')
-def post(data: POSTEmail):
+def send_emails(
+    data: POSTEmail,
+    background_tasks: BackgroundTasks,
+    email_client: EmailClient = Depends(get_email_client),
+):
```
Comment (afmam) [inline notification/components/email/views.py:46]: Missing docstring and return type hint

**Review by afmam — CHANGES_REQUESTED** [review]
Left a few comments

File: notification/components/email/email_client.py [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+                attach = MIMEImage(attach_data)
+                attach.add_header('Content-Disposition', 'attachment', filename=filename)
+            else:
+                attach = MIMEApplication(attach_data, _subtype='pdf', filename=filename)
+                attach.add_header('Content-Disposition', 'attachment', filename=filename)
+            return code, result, attach
+
+    def send_emails(self, receivers, sender, subject, text, msg_type, attachments):
+        try:
+            client = smtplib.SMTP(self.host, self.port)
```
[the remark this answers is not in the document]
Comment (vmoshynskyi) [reply notification/components/email/email_client.py:121]: Yes, that makes sense! I will keep this in mind while refactoring. Thank you!

File: notification/components/email/schemas.py (hunk 3 above)
Comment (vmoshynskyi) [reply notification/components/email/schemas.py:35]: This schema will be removed.

File: notification/components/email/schemas.py (hunk 2 above)
Comment (vmoshynskyi) [reply notification/components/email/schemas.py]: I've updated this schema.

File: notification/components/email/views.py (hunk 4 above)
Comment (vmoshynskyi) [reply notification/components/email/views.py]: I've fixed this.

**Review by afmam — APPROVED** [review]
LGTM

**Pull request merged**